src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -175,14 +193,14 @@ ov::Tensor LazyTensorImpl::eval() const {
 
     // Process the initial tensor - either from Const or from Concat
     if (!m_parent) {
-        if (m_transform.first == TransformType::THIS) {
-            return get_orig_tensor();
+        if (m_transform.first == TransformType::TENSOR) {
+            return std::get<ov::Tensor>(m_transform.second);
+        } else if (m_transform.first == TransformType::CONST) {
+            return ov::npuw::util::tensor_from_const(std::get<ConstPtr>(m_transform.second));
```

Comment:
No way, first of all, you don't need a parent. You can have multiple of, what's the parent in case of unpack? concat?

Second, as your Transform is a variant, just use the standard `std::visit` so dispatch your evaluation code based on the type. Will make this code much easier.

Reply:
Done